packages/terra-notification-dialog/src/NotificationDialog.jsx

```diff
@@ -112,179 +103,119 @@ const propTypes = {
-      : <Button text={rejectAction.text} onClick={rejectAction.onClick} />;
+    const buttonVariant = emphasizedAction === 'accept' ? { variant: ButtonVariants.EMPHASIS } : {};
+    actionButtons.push(<Button {...buttonVariant} data-terra-notification-dialog-button="accept" key="accept" text={acceptAction.text} onClick={acceptAction.onClick} />);
   }
 
-  if (buttonOrder === 'rejectFirst') {
-    return (
-      <div className={cx('actions')}>
-        {rejectButton}
-        {acceptButton}
-      </div>
-    );
+  if (rejectAction) {
+    const buttonVariant = emphasizedAction === 'reject' ? { variant: ButtonVariants.EMPHASIS } : {};
+    actionButtons.push(<Button {...buttonVariant} data-terra-notification-dialog-button="reject" key="reject" text={rejectAction.text} onClick={rejectAction.onClick} />);
   }
 
   return (
     <div className={cx('actions')}>
-      {acceptButton}
-      {rejectButton}
+      {buttonOrder === 'acceptFirst' ? actionButtons : actionButtons.reverse()}
     </div>
   );
 };
 
-const getIcon = (variant, customIcon = null) => {
-  switch (variant) {
-    case variants.ALERT:
-      return (<span className={cx(['icon', 'alert'])} />);
-    case variants.ERROR:
-      return (<span className={cx(['icon', 'error'])} />);
-    case variants.WARNING:
-      return (<span className={cx(['icon', 'warning'])} />);
-    case variants.INFO:
-      return (<span className={cx(['icon', 'info'])} />);
-    case variants.SUCCESS:
-      return (<span className={cx(['icon', 'success'])} />);
-    case variants.CUSTOM:
-      return customIcon;
-    default:
-      return null;
+const NotificationDialog = (props) => {
+  const theme = React.useContext(ThemeContext);
+
+  const {
+    title,
+    startMessage,
+    endMessage,
+    content,
+    acceptAction,
+    rejectAction,
+    variant,
+    isOpen,
+    buttonOrder,
+    emphasizedAction,
+    custom,
+    ...customProps
+  } = props;
+
+  if (isOpen && acceptAction === undefined && rejectAction === undefined) {
+    throw new Error('Unable to render the Notification dialog. Neither an `acceptAction` and/or `rejectAction` prop was provided which leaves the user without the ability to close the dialog.');
   }
-};
-
-class NotificationDialog extends React.Component {
-  constructor(props) {
-    super(props);
 
-    this.handleKeydown = this.handleKeydown.bind(this);
-    this.escapeKey = KeyCode.KEY_ESCAPE;
+  if (isOpen && !isValidVariant(variant)) {
+    throw new Error('Unable to render the Notification dialog as the `variant` prop was not provided.');
   }
 
-  componentDidMount() {
-    document.addEventListener('keydown', this.handleKeydown);
-  }
-
-  componentWillUnmount() {
-    document.removeEventListener('keydown', this.handleKeydown);
-  }
+  const signalWord = variant === variants.CUSTOM ? (custom || {}).signalWord : <FormattedMessage id={`Terra.notification.dialog.${variant}`} />;
```

Comment:
Why not set `custom` to `{}` in defaultProps instead of inline?

Reply:
good point. Not sure why I did this

Reply:
fixed: https://github.com/cerner/terra-framework/pull/1105/commits/022584400b4aaafcf4ad53ff1ed2e2b2eaec08a5#diff-cb62dc7ce0197bc569a9952450bb9fe5R103